packages/liveblocks-core/src/lib/__tests__/EventSource.test.ts

```diff
@@ -154,4 +193,58 @@ describe("EventSource", () => {
       )
     );
   });
+
+  it("awaiting events", async () => {
+    const src = makeEventSource();
+    const promise$ = src.waitUntil();
+
+    // Now notify, so the promise will resolve
+    src.notify(0);
+    src.notify(1);
+    src.notify(2);
+
+    await expect(promise$).resolves.toBe(0);
+  });
+
+  it("awaiting events conditionally", async () => {
+    const src = makeEventSource<number>();
+    const promise$ = src.waitUntil((n) => n % 2 === 1);
+
+    // Now notify, so the promise will resolve
+    src.notify(2);
+    src.notify(4);
+    src.notify(6);
+    src.notify(7); // First odd number, so we'll wait until this one!
+    src.notify(8);
+
+    await expect(promise$).resolves.toBe(7);
+  });
+
+  it("pausing/continuing event delivery", () => {
+    fc.assert(
+      fc.property(
+        anything(),
+
+        (payload) => {
+          const callback = jest.fn();
+          const hub = makeEventSource();
+
+          const unsub = hub.observable.subscribe(callback);
+
+          hub.pause();
+          hub.notify(payload);
+          hub.notify(payload);
+          hub.notify(payload);
+
+          expect(callback.mock.calls.length).toBe(0); // No events get delivered until unpaused
+
+          hub.unpause();
+          expect(callback.mock.calls.length).toBe(3); // Buffered events get delivered
```

Comment:
Very minor nitpick (or question in case you explicitly want `mock.calls.length`)
 
```suggestion
          expect(callback).not.toHaveBeenCalled(); // No events get delivered until unpaused

          hub.unpause();
          expect(callback).toHaveBeenCalledTimes(3); // Buffered events get delivered
```

Reply:
Oh the rest of the test uses `mock.calls.length`, never mind ignore this 😅

Reply:
Fixed in b0961257b12e9ecdb28d689d69c5cdee3d4e5f6e 👍